---
title: "Comparative genomic analysis of Flavobacteriaceae: insights into carbohydrate
  metabolism, gliding motility and secondary metabolite biosynthesis"
author: "Asimenia Gavrillidou"
date: "February 10, 2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Analysis of Flavobacteriaceae whole genomes sequencing data:

Raw reads of Flavobacteriaceae genomes sequenced in this study were deposited at the the European Nucleotide Archive (ENA) under Study Accession Number PRJEB35092.

The complete workflow:

    1. Trimming and Adapter removal of raw reads
    2. Genome Assembly and Quality control
    3. Genome annotation

##Trimming and Adapter removal of raw reads

Analysis was performed in a Linux environment using the local server of the Laboratory of Microbiology at Wageningen University & Research.

FASTQC 0.11.4 was used to check the quality of the raw reads.

Quality filtering was performed by Trimmomatic 0.32

To improve the per base sequence quality, 10 bases were removed from the beggining and the end of the reads.
Reads shorter than 20 bases were discarded.
A .fasta file (adapters.fa) including all Truseq adapter sequences found in the reads as overrepresented sequences was used to remove them from the raw reads.


java -jar ~/trimmomatic-0.32.jar PE -threads 12 -phred33 ~/RawReads2_1.fastq ~/RawReads2_2.fastq ~/R1.paired.fq ~/R1.unpaired.fq ~/R2.paired.fq ~/R2.unpaired.fq ILLUMINACLIP:~/adapters.fa:2:30:10 HEADCROP:15 CROP:100 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:20 MINLEN:50


After quality filtering, the reads were checked again with FASTQC. No additional quality filtering was needed.

##Genome Assembly and Quality Control

For the MiSeq data, A5 assembler was used:

a5_pipeline.pl --end=5 <paired.1.fastq> <paired.2.fastq> <assembly_A5>

For the Hiseq data, best k-mer size was predicted (KmerGenie 1.6741) and then assembly with SPAdes followed. 

~/kmergenie-1.6741/kmergenie ~/RawReads1_2.fastq
~/SPAdes-3.11.1-Linux/bin/spades.py -1 <R1.paired.fq> -2 <R1.paired.fq> --continue --careful -k 55 -o ~/Assembly_SPAdes

For both sequence reads, the online version of BLASTn (default settings) was used to identify potential contamination in the reads.

All contigs that matched to a contaminant sequence on NCBI were manually removed from the assemblies.

Output file: **Assembly_clean.fasta**


