---
title: "Comparative genomic analysis of Flavobacteriaceae: insights into carbohydrate metabolism, gliding motility and secondary metabolite biosynthesis"
author: "Asimenia Gavrillidou"
date: "February 10, 2020"
output: html_document
---

#Analysis of Flavobacteriaceae whole genomes sequencing data:

Raw reads of Flavobacteriaceae genomes sequenced in this study were deposited at the the European Nucleotide Archive (ENA) under Study Accession Number PRJEB35092.

The complete workflow:

    1. Trimming and Adapter removal of raw reads
    2. Genome Assembly and Quality control
    3. Genome annotation
    4. Phylogenomic Analysis

##Trimming and Adapter removal of raw reads

Analysis was performed in a Linux environment using the local server of the Laboratory of Microbiology at Wageningen University & Research.

FASTQC 0.11.4 was used to check the quality of the raw reads.

Quality filtering was performed by Trimmomatic 0.32

To improve the per base sequence quality, 10 bases were removed from the beggining and the end of the reads.
Reads shorter than 20 bases were discarded.
A .fasta file (adapters.fa) including all Truseq adapter sequences found in the reads as overrepresented sequences was used to remove them from the raw reads.

java -jar ~/trimmomatic-0.32.jar PE -threads 12 -phred33 ~/RawReads2_1.fastq ~/RawReads2_2.fastq ~/R1.paired.fq ~/R1.unpaired.fq ~/R2.paired.fq ~/R2.unpaired.fq ILLUMINACLIP:~/adapters.fa:2:30:10 HEADCROP:15 CROP:100 LEADING:20 TRAILING:20 SLIDINGWINDOW:4:20 MINLEN:50

After quality filtering, the reads were checked again with FASTQC. No additional quality filtering was needed.

##Genome Assembly and Quality Control

For the MiSeq data, A5 assembler was used:

a5_pipeline.pl --end=5 <paired.1.fastq> <paired.2.fastq> <assembly_A5>

For the Hiseq data, best k-mer size was predicted (KmerGenie 1.6741) and then assembly with SPAdes followed. 

~/kmergenie-1.6741/kmergenie ~/RawReads1_2.fastq
~/SPAdes-3.11.1-Linux/bin/spades.py -1 <R1.paired.fq> -2 <R1.paired.fq> --continue --careful -k 55 -o ~/Assembly_SPAdes

For both sequence reads, the online version of BLASTn (default settings) was used to identify potential contamination in the reads.

All contigs that matched to a contaminant sequence on NCBI were manually removed from the assemblies.

Output file: **Assembly_clean.fasta**

###Draft assembly improvement (Pilon 1.22)

Create a .bam file of quality-filtered reads aligned to the input genome:
bowtie2-build <Assembly_clean.fasta> <indexed.genome>
bowtie2-align -x <indexed.genome> -1 <R1.paired.fq> -2 <R2.paired.fq> -S <Assembly.sam> -t -p 4

Convert .sam file to .bam file:
samtools view -S <Assembly.sam> -b -o <Assembly.bam>

Sort and index .bam file:
samtools sort <Assembly.bam> <Assembly_sorted>
samtools index <Assembly_indexed_sorted> 
  
Run Pilon:
java -Xmx32G -jar ~/pilon-1.22.jar --genome Assembly_clean.fasta --frags Assembly.sorted.bam --outdir Pilon

Output file: **Assembly_clean_Pilon.fasta**

###Coverage per base calculation

bedtools genomecov -d -ibam Assembly.sorted.bam -g Assembly_clean_Pilon.fasta > Assembly_coverage.txt
awk '{sum+=$3}END{print sum/NR}' Assembly_coverage.txt

###Quality control of draft assemblies

~quast-4.6.3/quast.py Assembly_clean_Pilon.fasta -o ~/Quast

###Completeness and Contamination of draft assemblies (CheckM)
Before the comparative genomics analysis, we checked the completeness and contamination of all draft genomes (newly-sequenced and publicly available).

Taxonomic-specific 
checkm taxonomy_wf family Flavobacteriaceae -t 12 -x fasta ~/checkm_bins ~/Checkm_output_Taxonomic-specific

Lineage-specific
checkm lineage_wf -t 12 -x fasta ~/checkm_bins ~/Checkm_output

#Genome Annotation

##Annotation of CAZymes for all genomes included in the study based on http://bcb.unl.edu/dbCAN2/download/Databases/dbCAN-old@UGA/readme.txt 

###Predict protein sequences
/home/gavri003/prokka/bin/prokka --outdir <Prokka_annotations> --force --centre 20 --locustag <Assembly_Clean_Pilon.fasta>
  
Output file: **Prokka.faa**

###Download dbCAN CAZyme domain HMM database
wget http://csbl.bmb.uga.edu/dbCAN/download/dbCAN-fam-HMMs.txt

###Prepare the HMM database for scan
hmmpress dbCAN-fam-HMMs.txt

###Scan the database
hmmscan --domtblout Summary_Cazymes.out.dm ~/databases/dbCAN-fam-HMMs.txt Prokka.faa > Cazymes.out

###Remove overlapping hits (if alignment > 80aa, use E-value < 1e-5, otherwise use E-value < 1e-3; covered fraction of HMM > 0.3)
sh hmmscan-parser.sh Summary_Cazymes.out.dm > Cazymes.out.dm.ps

###Write the output in a .txt file to open with R
cat Cazymes.out.dm.ps > Cazymes.txt 

For each genome we had a CAZyme annotation file. 
Relative abundance of CAZyme genes (normalized by genome size) per genome between groups and clades were further calculated in Excel.

Output file: **Mean_Rel_Abund_Cazymes.xlsx**

##Annotation of PULs for all genomes included in the analysis using Open Prediction of Polysaccharide Utilization Loci tool (PULPy) 

To install, follow guidelines at: https://github.com/WatsonLab/PULpy

Create a directory called "genomes" with final format and unique ID followed by "genome_name_genomic.fna.gz"

To run it

snakemake snakefile --use-conda --cores 4

Output files: **genome_puls_sums.tsv** and **genome_puls.tsv**

#Phylogenomic Analysis

##Annotation of single-copy marker genes with GTDBTk

To install, follow guidelines at:https://ecogenomics.github.io/GTDBTk/running/

###Identify single-copy marker genes in the genomes

gtdbtk identify --genome_dir genomes/ --out_dir identify_output --cpus 4

###Align identified marker genes without including GTDB reference genomes in the multiple sequence alignment

gtdbtk align --identify_dir identify_output/ --out_dir align_output --skip_gtdb_refs --cpus 4

Output file: **prefix_user_msa.fasta**

##Build single-copy marker gene phylogenetic tree using FastTree 

To install, follow guidelines at: http://www.microbesonline.org/fasttree/#Install

FastTree prefix_user_msa.fasta > prefix_genome_tree.tree

Output file: **prefix_genome_tree.tree**

Use the output file as input for iTOL (https://itol.embl.de/)

